From: Maximilian Weigand <mweigand@mweigand.net>
Date: Sun, 4 Jun 2023 20:50:13 +0200
Subject: drm/rockchip: ebc: modify panel timings for the eink_ed103tc2 panel

* horizontal timings were checked against the data sheet, and a possible
  slight mistake was found. However, should be checked again and this
  change did not yield any visible changes to display behavior
* the source clock was reduced from effectively 250 MHz to 200 MHz
  (rounded values actually enabled). This change leads to a reduction of
  reported display refresh rate from 85 Hz to 64.74 Hz. This change is
  not notable even in A2 usage, but yields significantly better artifact
  behavior. Scenarios that previously yielded bad artifacting include
  xournalpp and evince scrolling in A2/A2+dithering mode when windowed
  (i.e., when the apps take up only half the screen space).
---
 drivers/gpu/drm/panel/panel-simple.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 858aaa1..f86b5832 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2096,11 +2096,18 @@ static const struct panel_desc edt_etmv570g2dhu = {
 };
 
 static const struct drm_display_mode eink_ed103tc2_mode = {
-	.clock = 266693,
+	/* this will lead to dclk == 250 Mhz */
+	/* .clock = 266693, */
+	/* request dclk == 200 MHz */
+	.clock = 200000,
 	.hdisplay = 1872,
-	.hsync_start = 1872 + 184,
-	.hsync_end = 1872 + 184 + 88,
-	.htotal = 1872 + 184 + 88 + 64,
+	/* hsync_start, for linux timing order, should be display + fp
+	 * according to the data sheet, display = ldl == 234 [dclk] = 1872 pixels, and fp == lel == 7 [dclk] == 56 pixels
+	 */
+	.hsync_start = 1872 + 56,
+	/* data sheet days sync == lsl == 18 sclk = 18 * 8 pixels = 144 pixels */
+	.hsync_end = 1872 + 56 + 144,
+	.htotal = 1872 + 56 + 144 + 136,
 	.hskew = 136,
 	.vdisplay = 1404,
 	.vsync_start = 1404 + 12,
